### DISCUSSION TITLE
Add NPM packages from Meteor packages to package-lock or create a top-level lockfile

### DISCUSSION THREAD

**Discussion by jankapunkt** [meteor/meteor#11696] [Feature Requests]
I just realized, that I can't let my dependabot and security actions analyze dependencies when they were declared in packages via `Npm.depends` and the package is not part of my source (for instance added from atmosphere).

I think this is something very important, because these are "hidden" security issues that we can't directly detect without an expensive manual audit of all the Meteor packages and what NPM dependencies they declared.

**Comment by filipenevola** [comment]
Hi @jankapunkt, I was checking what we already have and maybe we are not that far.

I see that for local packages we have npm-shrinkwrap.json inside `.meteor/local/isopacks/packageName/npm/node_modules`:

![galaxy-server – ~:Documents:meteor:ws:galaxy-server: meteor:local:isopacks:admin-accounts:npm:node_modules: npm-shrinkwrap json 2021-10-15 09-26-37](https://user-images.githubusercontent.com/437372/137495349-53ee4f00-4dd7-4409-9fe3-a4d5777e2e5d.png)

We would need somehow to also generate these files for packages from Atmosphere.

And maybe put the files in a better location inside `.meteor` folder.

What do you think? Would this be a solution?

**Comment by CaptainN** [comment]
Better if we deprecate atmosphere packages and move things to npm.
